airflow/jobs.py

```diff
@@ -509,7 +508,7 @@ def process_dag(self, dag, executor):
             if ti.state in (
                     State.RUNNING, State.QUEUED, State.SUCCESS, State.FAILED):
                 continue
-            elif ti.is_runnable(flag_upstream_failed=True):
+            elif ti.are_dependencies_met(flag_upstream_failed=True):
```

Comment:
Please provide the session if not necessary to have a new session. It reduces connection overhead.